inc/common.php

```diff
@@ -1273,19 +1269,19 @@ function detectExternalEdit($id) {
     global $lang;
 
     $fileLastMod = wikiFN($id);
-    $lastMod     = @filemtime($fileLastMod); // from page
+    $lastMod     = @filemtime($fileLastMod); // from wiki page, suppresses warning in case the file not exists
     $pagelog     = new PageChangeLog($id, 1024);
     $lastRev     = $pagelog->getRevisions(-1, 1); // from changelog
     $lastRev     = (int) (empty($lastRev) ? 0 : $lastRev[0]);
 
-    if(!file_exists(wikiFN($id, $lastMod)) && file_exists($fileLastMod) && $lastMod >= $lastRev) {
+    if (!file_exists(wikiFN($id, $lastMod)) && file_exists($fileLastMod) && $lastMod >= $lastRev) {
         // add old revision to the attic if missing
         saveOldRevision($id);
         // add a changelog entry if this edit came from outside dokuwiki
-        if($lastMod > $lastRev) {
+        if ($lastMod > $lastRev) {
             $fileLastRev = wikiFN($id, $lastRev);
             $revinfo = $pagelog->getRevisionInfo($lastRev);
-            if(empty($lastRev) || !file_exists($fileLastRev) || $revinfo['type'] == DOKU_CHANGE_TYPE_DELETE) {
+            if (empty($lastRev) || !file_exists($fileLastRev) || $revinfo['type'] == DOKU_CHANGE_TYPE_DELETE) {
```

Comment:
@Klap-in Good suggestion! I am now considering the way to register an external edit as revision in PageChangeLog::getCurrentRevisionInfo().

Reply:
See also example 
https://github.com/splitbrain/dokuwiki/commit/da59b6eea29a714bd94d93bce722e727b4c831ed
in https://github.com/splitbrain/dokuwiki/commits/revdiffexternaledits in that branch are also more unit tests.

The reason that external edits are registered there, is because external edits can continue happening multiple times after each other. So if you start registering every time an random person view the list with revisions/diff view, you will get a lot of saved external edits. Therefore, saving external edits are delayed as long as possible, that is up to the moment an normal edit happens.

Reply:
@Klap-in updated `detectExternalEdit()` and unittest `common_saveWikiText.test.php` file using currentRevisionInfo approach. I appreciate it if you could tell me some missing scenario or save-sequence in the unit tests.

Reply:
My check list for the external edits in the unit tests was:
- external created -> edit
- created -> external edit ->  edit
- created -> external delete -> create
- edit -> external edit -> edit
- edit -> external delete (covered by 3.3->3.4)
- deleted -> external create -> edit (covered by 2.2->2.3->2.4)
- reverted -> external edit ->edit (covered by 1.7->1.8->1.9)

3.1->3.2->3.3 (external create ->external edit->edit) is different than above 

If I looked correctly, now four save-sequences are not covered in the unit test.

Reply:
- [ ] external created -> edit
- [ ] created -> external edit -> edit
- [ ] created -> external delete -> create
- [ ] edit -> external edit -> edit

Reply:
@Klap-in Thanks reviewing the long lines of `common_saveWikiText.test.php` file. all done.